cli/generate.go

```diff
@@ -0,0 +1,279 @@
+/*
+Copyright 2022 Nethermind
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cli
+
+import (
+	"fmt"
+	"github.com/NethermindEth/sedge/cli/actions"
+	"github.com/NethermindEth/sedge/configs"
+	"github.com/NethermindEth/sedge/internal/pkg/clients"
+	"github.com/NethermindEth/sedge/internal/pkg/generate"
+	"github.com/NethermindEth/sedge/internal/ui"
+	"github.com/NethermindEth/sedge/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+// Global vars
+var (
+	generationPath string
+	network        string
+	logging        string
+)
+
+// GenCmdFlags is a struct that holds the flags of the generate command
+type GenCmdFlags struct {
+	CustomFlags
+	executionName     string
+	consensusName     string
+	validatorName     string
+	checkpointSyncUrl string
+	feeRecipient      string
+	noMev             bool
+	mevImage          string
+	mevBoostOnVal     bool
+	noValidator       bool
+	jwtPath           string
+	graffiti          string
+	mapAllPorts       bool
+	fallbackEL        *[]string
+	elExtraFlags      *[]string
+	clExtraFlags      *[]string
+	vlExtraFlags      *[]string
+	relayURL          string
+	mevBoostUrl       string
+	executionApiUrl   string
+	executionAuthUrl  string
+	consensusApiUrl   string
+}
+
+func GenerateCmd(sedgeAction actions.SedgeActions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate [flags]",
+		Short: "Generate new setups according to selected options",
+		Long: `Generate new setups according to selected options and flags.
+
+It will create a 'docker-compose.yml' and an '.env', you will need later to run the docker-compose script.
+
+You can generate:
+- Full Nodes (consensus+execution+validator)
+- Full Nodes without Validator (consensus+execution)
```

Comment:
```suggestion
It will create a 'docker-compose.yml' and a '.env', which you will need later to run the docker-compose script. You can use `sedge run` to run the script and start the setup.

You can generate:
- Full Nodes (consensus + execution + validator)
- Full Nodes without Validator (consensus + execution)
```

Reply:
Done at faacf787d646dade513226be2384704e0e6f9ad9